lib/notifications/notification_server.ex

```diff
@@ -36,6 +36,29 @@ defmodule Notifications.NotificationServer do
     GenServer.cast(server, {:bridge_movement, bridge_movement})
   end
 
+  @doc """
+  Creates a new Detour Activated Notification for a `Skate.Detours.Db.Detour`.
+
+  ## Options
+
+  If the `:server` option is present, the notification is sent to the process
+  referred to by the `:server` value.
+
+  If the `:notify_finished` option is present, a `{:new_notification, detour: detour.id}` message
+  is sent to the process referred to by the `:notify_finished` value.
```

Comment:
question: I don't quite see the situations where these options would be provided, and I checked your subsequent PR and don't see examples there either. Why do detours need these additional options, but the other notifications don't? (Not a request for more documentation. I'm glad these comments are here, but I still just don't understand the big picture.)

Reading some of the later comments: is :notify_finished partly included to support testing?

Reply:
So I originally had this function written as `detour/3` => `detour(detour, server, notify_finished)`

This followed the other "new notification" functions but with a new `notify_finished` argument. As I was starting to work this this, I found myself not certain about the order of arguments (should `server` come first because it may be used more? or will `notify_finished` be used more?) That pointed me to think that I was actually approaching this wrong and the other signatures are what should be changed to use keyword lists, so I went with keywords. I struggled with _when_ to change the other signatures though, which is why I've only made this change for detours :/ .

But that said, I'm also not entirely happy with the `notify_finished` argument either. It was only useful for the tests for `notification_server` despite me _wishing_ it was useful for more tests, like the [`detours_controller` tests](https://github.com/mbta/skate/pull/2821/files#diff-6b5f7db841fd66bb4a09b0870661a72d957ad6cf263cccf090ebb4a251bc0687). I couldn't find a way to nicely inject that argument into the call for tests, and I also haven't found any places in the code where `notify_finished` useful for non-testing code, which is frustrating.
The only thing which made me think that it's worth keeping around, is that the `Skate.Detours.TripModificationPublisher`'s `:on_connect` option is essentially exactly the same as the `notify_finished` argument, and has also not been useful outside of tests, but has been a needed and elegant solution to the issue of waiting for an event to happen before continuing ([tests]; but ideally, maybe, in the future, additionally non-test code).

Reply:
Thank you so much!! Amazing and thoughtful reply. I like knowing it's primarily a testing argument right now, and I'm ok with having it in the hopes it will prove helpful for some future non-test code. But even if we don't, it's valuable to have it for testing.

Reply:
I used to have a comment along these lines, but I dropped it apparently, thoughts?
```suggestion
  If the `:server` option is present, the notification is sent to the process
  referred to by the `:server` value.

  If the `:notify_finished` option is present, a `{:new_notification, detour: detour.id}` message
  is sent to the process referred to by the `:notify_finished` value.
  This option has mainly been useful for testing code to avoid `Process.sleep()` calls.
```

Reply:
Oh yes, I do like that final line for clarity! Thanks